roles/matrix-bot-matrix-registration-bot/templates/systemd/matrix-bot-matrix-registration-bot.service.j2

```diff
@@ -0,0 +1,38 @@
+#jinja2: lstrip_blocks: "True"
+[Unit]
+Description=Matrix registration bot
+{% for service in matrix_bot_matrix_registration_bot_systemd_required_services_list %}
+Requires={{ service }}
+After={{ service }}
+{% endfor %}
+{% for service in matrix_bot_matrix_registration_bot_systemd_wanted_services_list %}
+Wants={{ service }}
+{% endfor %}
+DefaultDependencies=no
+
+[Service]
+Type=simple
+Environment="HOME={{ matrix_systemd_unit_home_path }}"
+ExecStartPre=-{{ matrix_host_command_sh }} -c '{{ matrix_host_command_docker }} kill matrix-bot-matrix-registration-bot 2>/dev/null || true'
+ExecStartPre=-{{ matrix_host_command_sh }} -c '{{ matrix_host_command_docker }} rm matrix-bot-matrix-registration-bot 2>/dev/null || true'
+
+ExecStart={{ matrix_host_command_docker }} run --rm --name matrix-bot-matrix-registration-bot \
+			--log-driver=none \
+			--cap-drop=ALL \
+			--network={{ matrix_docker_network }} \
+			-e 'BOT_SERVER=https://{{ matrix_server_fqn_matrix }}' \
+			-e 'BOT_USERNAME={{ matrix_bot_matrix_registration_bot_matrix_user_id_localpart }}' \
+			-e 'BOT_PASSWORD={{ matrix_bot_matrix_registration_bot_matrix_user_password }}' \
+			-e 'API_BASE_URL=https://{{ matrix_server_fqn_matrix }}' \
+			-e 'API_TOKEN={{ matrix_bot_matrix_registration_bot_matrix_admin_token }}' \
+			-e 'LOGGING_LEVEL=info' \
```

Comment:
Looks like you've decided to configure the both with environment variables, not with a `config.yml` file.

This is fine, but it's better (and safer) to not use `-e` here, but to pass an environment variables file using `--env-file`. Doing it via a file helps keep the password hidden from system users doing `ps`, etc. It's also a cleaner way to manage a bunch of environment variables.

To see how to use an `--env-file`, see how the `matrix-bot-honoroit` role does it.

Also note its usage of the `matrix_bot_honoroit_environment_variables_extension` variable, which allows users to inject additional variables, besides the one we define explicitly.

I think you should also have a dedicated variable for each of this environemnt variables.

Instead of defining `BOT_SERVER` as `https://{{ matrix_server_fqn_matrix }}`, it'd be better to have a `matrix_bot_matrix_registration_bot_bot_server` variable that is defined like this in `defaults/main.yml`:

```yaml
matrix_bot_matrix_registration_bot_bot_server: "https://{{ matrix_server_fqn_matrix }}"
```

This allows advanced users to override the whole `matrix_bot_matrix_registration_bot_bot_server` variable, should they need to.

The same goes for the other environment variables.

------

`LOGGING_LEVEL` seems completely hardcoded. You may wish to add a variable (`matrix_bot_matrix_registration_bot_logging_level: info`) for this one as well.